lib/wifi_user/use_case/email_signup.rb

```diff
@@ -2,30 +2,33 @@
 require 'notifications/client'
 
 class WifiUser::UseCase::EmailSignup
-  def initialize(user_model:)
+  def initialize(user_model:, whitelist_checker:, logger: Logger.new(STDOUT))
     @user_model = user_model
+    @whitelist_checker = whitelist_checker
+    @logger = logger
   end
 
   def execute(contact:)
     email_address = Mail::Address.new(contact).address
 
-    return unless Common::EmailAddress.authorised_email_domain?(email_address)
-
-    login_details = user_model.generate(contact: email_address)
-    send_signup_instructions(email_address, login_details)
+    if whitelist_checker.execute(email_address)[:success]
+      send_signup_instructions(email_address)
+    else
+      logger.info("Unsuccessful email signup attempt: #{email_address}")
```

Comment:
Is it weird that when a user doesn't have a whitelisted e-mail, we don't reply to them with any help?

That's a feature request though, not for this PR - just thinking out loud.

Reply:
Yeah getting numbers on this feels like a logical first step